pkg/cache/mock_artifact_cache_types.go

```diff
@@ -0,0 +1,150 @@
+package cache
+
+import (
+	"github.com/stretchr/testify/mock"
+
+	"github.com/aquasecurity/trivy/pkg/fanal/types"
+)
+
+type ArtifactCacheDeleteBlobsArgs struct {
+	BlobIDs         []string
+	BlobIDsAnything bool
+}
+
+type ArtifactCacheDeleteBlobsReturns struct {
+	_a0 error
+}
+
+type ArtifactCacheDeleteBlobsExpectation struct {
+	Args    ArtifactCacheDeleteBlobsArgs
+	Returns ArtifactCacheDeleteBlobsReturns
+}
+
+func (_m *MockArtifactCache) ApplyDeleteBlobsExpectation(e ArtifactCacheDeleteBlobsExpectation) {
+	var args []any
+	if e.Args.BlobIDsAnything {
+		args = append(args, mock.Anything)
+	} else {
+		args = append(args, e.Args.BlobIDs)
+	}
+	_m.On("DeleteBlobs", args...).Return(e.Returns._a0).Maybe()
```

Comment:
`mockery` now checks that `call` was not called
But these are test cases where we don't get to that point (e.g. when we get an error earlier)

So we need to add `Maybe()` to skip that error.